[Bug]: semantic_ingestion failed with error asyncpg.exceptions.UniqueViolationError: duplicate key value violates unique constraint "citations_pkey"

### Describe the bug

Errors when semantic_memory processing one set_id, the traceback is:
```
2026-02-10 01:22:02,035 [ERROR] memmachine.semantic_memory.semantic_ingestion - Failed to process set_id mem_session_edu-org/study-buddy
Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 550, in _prepare_and_execute
    self._rows = deque(await prepared_stmt.fetch(*parameters))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/asyncpg/prepared_stmt.py", line 177, in fetch
    data = await self.__bind_execute(args, 0, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/asyncpg/prepared_stmt.py", line 268, in __bind_execute
    data, status, _ = await self.__do_execute(
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/asyncpg/prepared_stmt.py", line 257, in __do_execute
    return await executor(protocol)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "asyncpg/protocol/protocol.pyx", line 205, in bind_execute
asyncpg.exceptions.UniqueViolationError: duplicate key value violates unique constraint "citations_pkey"
DETAIL:  Key (feature_id, history_id)=(500, 110) already exists.

```

### Steps to reproduce

1. Adding memories into MemMachine
```
jinggong@Jings-MBP jgong % bash ingest_test_data.sh
Ingesting test data into http://localhost:8080/api/v2/memories ...

── Project 1: acme-corp / travel-assistant ──
  Sent 11 requests (33 messages)
── Project 2: acme-corp / health-coach ──
  Sent 12 requests (36 messages)
── Project 3: edu-org / study-buddy ──
  Sent 12 requests (36 messages)
```
2. Check container logs of memmachine-app, i can see the errors
```
2026-02-10 20:12:42,998 [INFO] httpx - HTTP Request: POST http://vmnet4-200.eng.memverge.com:30007/v1/embeddings "HTTP/1.1 200 OK"
2026-02-10 20:12:43,104 [INFO] httpx - HTTP Request: POST http://vmnet4-200.eng.memverge.com:30007/v1/embeddings "HTTP/1.1 200 OK"
2026-02-10 20:12:43,111 [INFO] memmachine.semantic_memory.semantic_ingestion - Finished processing 24 messages out of 24 for set mem_session_acme-corp/travel-assistant
2026-02-10 20:12:55,997 [INFO] httpx - HTTP Request: POST http://vmnet4-200.eng.memverge.com:30007/v1/chat/completions "HTTP/1.1 200 OK"
2026-02-10 20:12:56,187 [INFO] httpx - HTTP Request: POST http://vmnet4-200.eng.memverge.com:30007/v1/embeddings "HTTP/1.1 200 OK"
2026-02-10 20:12:56,194 [ERROR] memmachine.semantic_memory.semantic_ingestion - Failed to process set_id mem_session_acme-corp/travel-assistant
Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 550, in _prepare_and_execute
    self._rows = deque(await prepared_stmt.fetch(*parameters))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/asyncpg/prepared_stmt.py", line 177, in fetch
    data = await self.__bind_execute(args, 0, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/asyncpg/prepared_stmt.py", line 268, in __bind_execute
    data, status, _ = await self.__do_execute(
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/asyncpg/prepared_stmt.py", line 257, in __do_execute
    return await executor(protocol)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "asyncpg/protocol/protocol.pyx", line 205, in bind_execute
asyncpg.exceptions.UniqueViolationError: duplicate key value violates unique constraint "citations_pkey"
DETAIL:  Key (feature_id, history_id)=(87, 12) already exists.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 951, in do_execute
    cursor.execute(statement, parameters)
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 585, in execute
    self._adapt_connection.await_(
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 132, in await_only
    return current.parent.switch(awaitable)  # type: ignore[no-any-return,attr-defined] # noqa: E501
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 196, in greenlet_spawn
    value = await result
            ^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 563, in _prepare_and_execute
    self._handle_exception(error)
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 513, in _handle_exception
    self._adapt_connection._handle_exception(error)
  File "/app/.venv/lib/python3.12/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py", line 797, in _handle_exception
    raise translated_error from error
sqlalchemy.dialects.postgresql.asyncpg.AsyncAdapt_asyncpg_dbapi.IntegrityError: <class 'asyncpg.exceptions.UniqueViolationError'>: duplicate key value violates unique constraint "citations_pkey"
DETAIL:  Key (feature_id, history_id)=(87, 12) already exists.
```



### Expected behavior

Need to check if there is data integrity error, the exception can be handled properly

### Environment

- OS: MacOS
- Release: v0.2.6

### Additional context

_No response_

--- a/src/memmachine/semantic_memory/semantic_ingestion.py
+++ b/src/memmachine/semantic_memory/semantic_ingestion.py
@@ -344,7 +344,7 @@ class IngestionService:
             ],
         )
         citation_ids = TypeAdapter(list[EpisodeIdT]).validate_python(
-            list(merged_citations),
+            list(set(merged_citations)),
         )
 
         async def _add_feature(f: LLMReducedFeature) -> None:

